Automattic/jetpack
# Introduce Videopress Core JS Package

Creates the VideoPress Core package skeleton. 

The overall idea is to reuse existing [VideoPress JS library code](https://github.com/Automattic/jetpack/tree/trunk/projects/packages/videopress/src/client/lib/resumable-file-uploader) outside of the Jetpack/VideoPress codebase (for example, in themes built on top of Jetpack/VideoPress or integrations).

As a next step, the resumable uploader and the other related parts of the code will be moved to this package.
The uploader code "as-is", needs [get-media-token](https://github.com/Automattic/jetpack/tree/trunk/projects/packages/videopress/src/client/lib/get-media-token) and some other TS types.

Eventually, this would:

* Be deployed as an npm package.
* Contain/replace [videopress/lib](https://github.com/Automattic/jetpack/tree/trunk/projects/packages/videopress/src/client/lib).

## Proposed changes:

* None at this point.


## Jetpack product discussion

pxWta-1PG-p2

## Does this pull request change what data or activity we track or use?

No.

## Testing instructions:

* At this point this PR contains only the skeleton generated from `jetpack generate`, so I guess not much functionality to test. But maybe I missed something important on Generate.
* the build command should "work": `pnpm jetpack build --deps js-packages/videopress-core`
